### DISCUSSION TITLE
Using ConsoleLogger in a Laravel project

### DISCUSSION THREAD

**Discussion by sr-44** [nutgram/nutgram#606] [Q&A] [answered]
Is it possible to use `ConsoleLogger` in a Laravel project?
That is, when run the artisan `nutgram:run` command, I want to see the bot updates

**Comment by Lukasss93** [comment, marked as answer]
Yes, just do this:

You need to create the `nutgram` channel inside the `config/logging.php` file:

```php
'nutgram' => [
    'driver' => 'monolog',
    'level' => env('LOG_LEVEL', 'debug'),
    'handler' => StreamHandler::class,
    'formatter' => Nutgram\Laravel\Log\NutgramFormatter::class,
    'with' => [
        'stream' => 'php://stderr',
    ],
    'processors' => [PsrLogMessageProcessor::class],
],
```

Then add the `nutgram` channel to the `TELEGRAM_LOG_CHANNEL` env var:

```dotenv
TELEGRAM_LOG_CHANNEL=nutgram
```

#### Output:
![logger](https://i.imgur.com/SU7P8Ug.png)

**Reply by sr-44** [reply]
Undefined class `NutgramFormatter`

**Reply by Lukasss93** [reply]
Upgrade nutgram/laravel to 1.3.0

**Reply by azbagas** [reply]
I am using nutgram/laravel 1.3, but got this error:
`local.ERROR: Undefined array key "type" {"exception":"[object] (ErrorException(code: 0): Undefined array key \"type\" at C:\\laragon\\www\\bot-telegram\\vendor\\nutgram\\laravel\\src\\Log\\NutgramFormatter.php:13)`

But, when I delete the nutgram channel inside the config/logging.php file, everything seems fine.
